### PR TITLE
Add support to send mail to unauthenticated smtp relay

### PR DESCRIPTION
**Pull request by clarkchentw**
The current version only supports sending mail to an authenticated SMTP relay, which is not compatible with internal mail gateway e.g. Proxmox Mail Gateway. Adding this feature will allow configuring unauthenticated SMTP relay directly within the HestiaCP environment.

### CODE REVIEW

**Review by jaapmarcus — CHANGES_REQUESTED** [review]
@clarkchentw thank you for your contribution. 

Haven't tested it yet how ever with a quick look I noticed it is missing an upgrade path for current installs...

File: web/edit/server/index.php [hunk 1]
```diff
@@ -527,27 +527,22 @@
 
     // Update system wide smtp relay
     if (empty($_SESSION['error_msg'])) {
-        if (isset($_POST['v_smtp_relay']) && (!empty($_POST['v_smtp_relay_host'])) && (!empty($_POST['v_smtp_relay_user']))) {
+        if (isset($_POST['v_smtp_relay']) && !empty($_POST['v_smtp_relay_host'])) {
             if (($_POST['v_smtp_relay_host'] != $v_smtp_relay_host) ||
                      ($_POST['v_smtp_relay_user'] != $v_smtp_relay_user) ||
-                     ($_POST['v_smtp_relay_port'] != $v_smtp_relay_port) ||
```
Comment (jaapmarcus) [inline web/edit/server/index.php:582]: Port changes should still trigger v-add-sys-smtp-relay so is shouldn't be removed

File: web/edit/mail/index.php [hunk 2]
```diff
@@ -412,27 +412,22 @@
 
     // Add SMTP Relay Support
     if (empty($_SESSION['error_msg'])) {
-        if (isset($_POST['v_smtp_relay']) && (!empty($_POST['v_smtp_relay_host'])) && (!empty($_POST['v_smtp_relay_user']))) {
+        if (isset($_POST['v_smtp_relay']) && !empty($_POST['v_smtp_relay_host'])) {
             if (($_POST['v_smtp_relay_host'] != $v_smtp_relay_host) ||
                 ($_POST['v_smtp_relay_user'] != $v_smtp_relay_user) ||
-                ($_POST['v_smtp_relay_port'] != $v_smtp_relay_port) ||
```
Comment (jaapmarcus) [inline web/edit/mail/index.php:436]: Port changes should still trigger v-add-mail-domain-smtp-relay

**Review by jaapmarcus — CHANGES_REQUESTED** [review]
See additional comments

File: web/edit/server/index.php (hunk 1 above)
Comment (clarkchentw) [reply web/edit/server/index.php:582]: See next line as well, this change is to remove the requirement for empty smtp_relay_pass, the smtp_relay_port is still on the same line

File: web/edit/mail/index.php (hunk 2 above)
Comment (clarkchentw) [reply web/edit/mail/index.php:436]: This change is to remove the requirement for empty smtp_relay_pass, the smtp_relay_port is still kept on the same line

**Comment by jaapmarcus** [comment]
Merged main branch into and changed upgrade script to make it suitable for 1.6.0

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**